src/pytest_workflow/plugin.py

```diff
@@ -138,7 +131,9 @@ def run_workflow(self):
         else:
             # addfinalizer adds a function that is run when the node tests are
             # completed
-            self.addfinalizer(tempdir.remove)
+            def rm_tempdir():
+                shutil.rmtree(str(tempdir))
+            self.addfinalizer(rm_tempdir)
```

Comment:
Perhaps in stead of defining a function inside of a function (hairy imho) you could opt to use functools partial functions. E.g.

```python
from functools import partial

finalizer_func = partial(shutil.rmtree, str(tempdir))
self.addfinalizer(finalizer_func)
```

Reply:
:+1:
Great. I have seen this before and I believe this is the idiomatic way.